__tests__/ship.test.js

```diff
@@ -0,0 +1,293 @@
+        });
+
+        test('State of "_currentPort" should be null if "_itinerary.ports" is empty', () => {
+            const emptyConfig = {
+                _itinerary: {
+                    ports: []
+                }
+            };
+            const balticExplorer = new Ship(emptyConfig);
+            
+            expect(balticExplorer._currentPort).toBeNull();
+        });
+
+        test('State of "_isInitialized" should be false if "_itinerary.ports" is empty', () => {
+            const emptyConfig = {
+                _itinerary: {
+                    ports: []
+                }
+            };
+            const balticExplorer = new Ship(emptyConfig);
+
+            expect(balticExplorer._isInitialized).toBe(false);
+        });
+    });
+
+    describe('_setSail', () => {
+        let balticExplorerConfig;
+        let balticExplorer;
+        beforeEach(() => {
+            balticExplorerConfig = {
+                _itinerary: {
+                    ports: [
+                        {
+                            name: 'Helsinki',
+                            _ships: [],
+                            addShip: jest.fn(),
+                            removeShip: jest.fn()
+                        },
+                        {
+                            name: 'St Petersburg',
+                            _ships: [],
+                            addShip: jest.fn(),
+                            removeShip: jest.fn()
+                        }
+                    ]
+                }
+            };
+            balticExplorer = new Ship(balticExplorerConfig);
+        });
+
+        afterEach(() => {
+            balticExplorer = null;
+        });
+
+        test('Throws an error if "_itinerary.ports" is empty', () => {
+            const error = 'The ship has no itinerary.';
+
+            balticExplorer._itinerary.ports = [];
+            expect(() => {
+                balticExplorer._setSail();
+            }).toThrowError(error);
+        });
+
+        test('Throws an error if "_currentPort" is null', () => {
+            const error = 'The ship has already set sail.';
+
+            balticExplorer._currentPort = null;
+            expect(() => {
+                balticExplorer._setSail();
+            }).toThrowError(error);
+        });
+
+        test('Throws an error if "_itinerary.ports" has one port AND "_previouslyDockedPorts" is not empty', () => {
+            const error = 'The ship has already set sail.';
+
+            balticExplorer._itinerary.ports = [jest.fn()];
+            balticExplorer._previouslyDockedPorts.push(jest.fn());
+            expect(() => {
+                balticExplorer._setSail();
+            }).toThrowError(error);
+        });
+
+        test('State of "_currentPort" should be set to null', () => {
+            expect(balticExplorer._currentPort).not.toBeNull();
+            balticExplorer._setSail();
+            expect(balticExplorer._currentPort).toBeNull();
```

Comment:
The first assertion is testing something that hasnt been acted on since the setup, so will always return true and can be removed as it isnt adding any value to the test
```suggestion
            balticExplorer._setSail();

            expect(balticExplorer._currentPort).toBeNull();
```